src/ValueResult.cs

```diff
@@ -113,6 +113,11 @@ public new Type GetType()
         /// </summary>
         public static implicit operator bool(Result<TValue> result) => !result.IsError;
 
+        /// <summary>
+        /// Create an <see cref="List{IError}"/> from a result
+        /// </summary>
+        public static implicit operator List<IError>(Result<TValue> result) => result._errors;
```

Comment:
As discussed in our conversation, this is an amazing adition, but probably using and `IEnumerable<string>` or event `ICollection` would be better, so that people using it, can then use other collections that implement the interface, and have a even higher abstraction.
Let me know your feedback

Reply:
At the moment the conversion with implicit operator for a interface is not supported
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs0552?f1url=%3FappId%3Droslyn%26k%3Dk(CS0552)